include/ck_tile/ops/layernorm2d/kernel/layernorm2d_fwd_kernel.hpp

```diff
@@ -274,17 +234,176 @@ struct Layernorm2dFwd
         }
     }
 
+    template <typename XBlockWindow,
+              typename GammaBlockWindow,
+              typename BetaBlockWindow,
+              typename YBlockWindow,
+              typename MeanBlockWindow,
+              typename InvStdBlockWindow,
+              bool Cond = (kHasGamma && kHasBeta)>
+    CK_TILE_DEVICE std::enable_if_t<Cond>
+    OnePassLayernorm2dFwd(XBlockWindow& x_block_window,
+                          GammaBlockWindow& gamma_block_window,
+                          BetaBlockWindow& beta_block_window,
+                          YBlockWindow& y_block_window,
+                          MeanBlockWindow& mean_block_window,
+                          InvStdBlockWindow& inv_std_block_window,
+                          ComputeDataType epsilon,
+                          ck_tile::index_t N) const
+    {
+        // TODO: padding - handle max_count if N % kNPerBlock != 0
+        constexpr auto xDstr      = MakeXBlockTileDistribution();
```

Comment:
to reduce duplicated code, we may get distribution from `x_block_window`

Reply:
remove xDstr now